exercises/dolica_akelloegwel/ex02_oo_basics/src/ShapeSorter.h

```diff
@@ -1,17 +1,18 @@
 #pragma once
 
-#include <functional>
 #include "Shape.h"
+#include <memory>
+#include <functional>
 
 class ShapeSorter
 {
 public:
-	static bool AreaComparator(const Shape &s1, const Shape &s2);
-	static bool PerimeterComparator(const Shape &s1, const Shape &s2);
-	void PrintBasedOnType(std::vector<std::reference_wrapper<Shape>> shapes, std::string type);
-	void PrintBasedOnNSides(std::vector<std::reference_wrapper<Shape>> shapes, int numSides);
-	std::vector<std::reference_wrapper<Shape>> SortByAreaDesc(std::vector<std::reference_wrapper<Shape>> shapes);
-	std::vector<std::reference_wrapper<Shape>> SortByPerimeterDesc(std::vector<std::reference_wrapper<Shape>> shapes);
-	void PrintShapes(std::vector<std::reference_wrapper<Shape>> shapes);
+	static bool AreaComparator(const std::unique_ptr<Shape> &s1, const std::unique_ptr<Shape> &s2);
```

Comment:
1. I think that we should probably change all of these to be `std::unique_ptr<const Shape>` since we do not want to allow shape to be modified once created?
1. use of `std::unique_ptr<Shape>` (at present) is common and a little long-winded. how about declaring 
```using Shape_uptr = std::unique_ptr<Shape>;``` or ```using Shape_cuptr = std::unique_ptr<const Shape>;``` to clean up?